openshift_tools/monitoring/zbxapi.py

```diff
@@ -0,0 +1,256 @@
+                '''
+                if params:
+                    rpc_params['params'] = params
+                else:
+                    rpc_params['params'] = copy.deepcopy(rpc_params)
+
+                return self.parent.perform(cname.lower()+"."+method_name, rpc_params)
+
+            template_method.__doc__ = \
+              "https://www.zabbix.com/documentation/2.4/manual/api/reference/%s/%s" % \
+              (cname.lower(), method_name)
+            template_method.__name__ = method_name
+            # this is where the template method is placed inside of the subclass
+            # e.g. setattr(User, "create", stub_method)
+            setattr(_class, template_method.__name__, template_method)
+
+        # This class call instantiates a subclass. e.g. User
+        _class = type(cname,
+                      (object,),
+                      {'__doc__': \
+                      "https://www.zabbix.com/documentation/2.4/manual/api/reference/%s" % cname.lower()})
+        def __init__(self, parent):
+            '''
+            This init method gets placed inside of the _class
+            to allow it to be instantiated.  A reference to the parent class(ZabbixAPI)
+            is passed in to allow each class access to the perform method.
+            '''
+            self.parent = parent
+
+        # This attaches the init to the subclass. e.g. Create
+        setattr(_class, __init__.__name__, __init__)
+        # For each of our ZabbixAPI.classes dict values
+        # Create a method and attach it to our subclass.
+        # e.g.  'User': ['delete', 'get', 'updatemedia', 'updateprofile',
+        #                'update', 'iswritable', 'logout', 'addmedia', 'create',
+        #                'login', 'deletemedia', 'isreadable'],
+        # User.delete
+        # User.get
+        for method_name in method_names:
+            meta_method(_class, method_name)
+        # Return our subclass with all methods attached
+        return _class
+
+    def get_content(self, zbx_class_name, method, params):
+        '''
+        This bit of metaprogramming takes a zabbix_class_name (e.g. 'user' )
+        This gets the instantiated object of type user and calls method
+        with params as the parameters.
+
+        Returns the zabbix query results
+        '''
+        zbx_class_inst = self.__getattribute__(zbx_class_name.lower())
+        zbx_class = self.__getattribute__(zbx_class_name.capitalize())
+        return zbx_class.__dict__[method](zbx_class_inst, params)[1]
+
+
+# Attach all ZabbixAPI.classes to ZabbixAPI class through metaprogramming
+for _class_name, _method_names in ZabbixAPI.classes.items():
+    setattr(ZabbixAPI, _class_name, ZabbixAPI.meta(_class_name, _method_names))
+
+#if __name__ == '__main__':
+#    data = {
+#        'server': 'http://localhost/zabbix/api_jsonrpc.php',
+#        'username': '',
+#        'password': '',
+#    }
+#    zbx = ZabbixAPI(data)
+#    print zbx.get_content('user', 'get', {})
```

Comment:
Can these comments be removed?  Seems like dead code.

Or maybe they should be moved to the top as part of the module doc string to show how it can be used.